docs/webhook.asciidoc

```diff
@@ -0,0 +1,191 @@
+[id="{p}-webhook"]
+== Validating webhook
+
+A validating webhook provides additional validation of Elasticsearch resources beyond what can be specified in OpenAPI specs.
+This allows advanced validation to help prevent Elasticsearch resources that are invalid and will not function correctly
+from being admitted by the Kubernetes API server, which can aid troubleshooting.
+
+[float]
+=== Architecture
+The webhook is composed of 4 main components, here is a brief description of each of them to understand how they interact,
+their naming and how they are managed.
+
+. A `ValidatingWebhookConfiguration` object that defines the validating webhook. In ECK it is named `elastic-es-validation.k8s.elastic.co`. It must be created before to start the operator, the `caBundle` field can be automatically managed as part of the
+automatic certificate management _(see below)_.
+. A Kubernetes Service is used to expose the validating server, in ECK its name is fixed to `elastic-webhook-server`.
+It lives in the same Namespace that the webhook server.
+. A webhook server that actually validates the submitted resources. In ECK it is the operator itself when it
+is configured to act the `webhook` role. See <<{p}-operator-config,Configuring ECK>> if want to learn more about the
+`operator-roles` flag.
+. A Secret which contains the required certificates to secure the connection between the API server and the webhook.
+Like the ValidatingWebhookConfiguration it must be created before to start the operator, even if it is empty.
+If ECK has been installed with the provided manifest its name is `elastic-webhook-server-cert`.
+The content of this Secret and the lifecycle of the certificates are automatically handled for you. You can disable
+this feature if you want to manage the certificates by yourself or with the cert-manager, see an example below.
```

Comment:
...if you want to **set up**... see an example **for the latter** below ... (not 💯 about this last one)